src/main/scala/is/hail/methods/LDMatrix.scala

```diff
@@ -23,10 +24,15 @@ object LDMatrix {
     
     val nSamples = vds.nSamples
 
-    val filteredNormalizedHardCalls = vds.rdd.flatMap { 
-      case (v, (_, gs)) => RegressionUtils.normalizedHardCalls(gs, nSamples).map(x => (v, x))
-    }
-      .persist()
+    val rowType = vds.rowType
+
+    val filteredNormalizedHardCalls = vds.rdd2.mapPartitions { it =>
+      val view = HardCallView(rowType)
+      it.flatMap { r =>
+        val v = UnsafeRow.readStruct(rowType, r.region, r.offset).getAs[Variant](1)
+        RegressionUtils.normalizedHardCalls(r, view, nSamples).map(x => (v, x))
```

Comment:
You only use r in `normalizedHardCalls` to set the row.  So you could write:

```
  it.map { rv =>
    view.set(rv)
    RegressionUtils.normalizedHardCalls(view, nSamples)...
  }
```

Which is a bit tighter.